init commit (#234)
googleapis/python-storage: README_OPENTELEMETRY.rst

Metrics with OpenTelemetry
==================================
This library uses `OpenTelemetry <https://opentelemetry.io/>`_ to record metric information on API requests.
For information on the benefits and utility of tracing, see the `Cloud Monitoring docs <https://cloud.google.com/monitoring/docs>`_.

We first need to install opentelemetry:

.. code-block:: sh

>     pip install opentelemetry-api

Author: Yeah so opentelemetry is split into an api and sdk, and we'll need both (unless the user decides to make their own SDK). The opentelemetry-exporter-google-cloud is the package containining the Cloud Monitoring Exporter which we'll need as well. opentelemetry-tools-google-cloud is the package that contains the Cloud Resource Detector, which I suppose is optional. I'll mark it as such.

>     pip install opentelemetry-api

Reviewer: If you can relay, this is a large number of dependencies to add to a project. I think it may more than double the dependencies for this library.

Then, at the beginning of your application create a MeterProvider. Example:

.. code:: python

    # "opt-in" for opentelemtry metrics capturing.
    # Requires you to have installed opentelemetry packages and called set_meter_provider
    from opentelemetry import metrics
    from opentelemetry.sdk.metrics import MeterProvider

    #==============================================================================
    # OPTIONAL: These lines of code will detect + scrape resource info from the env/metadata.
    # This info will then be passed down the line and added to metric info
    from opentelemetry.tools.resource_detector import GoogleCloudResourceDetector
    resources = GoogleCloudResourceDetector().detect()

    # Note that this will NOT work if you are on a cloudtop, as you are on a
    # VM instacne that belongs to the project "cloudtop:prod" which you likely
    # don't have read access for
    #==============================================================================

    # If you don't use resources, remove resource=resources
    metrics.set_meter_provider(MeterProvider(stateful=False, resource=resources))

    # Normal code below. This is your application / script

> Check out captured metrics at `Cloud Monitoring <https://pantheon.corp.google.com/monitoring/metrics-explorer>`_.

Reviewer: change to external link

Metrics with OpenTelemetry
==================================
This library uses `OpenTelemetry <https://opentelemetry.io/>`_ to record metric information on API requests.

> For information on the benefits and utility of tracing, see the `Cloud Monitoring docs <https://cloud.google.com/monitoring/docs>`_.

Reviewer 2: nit: this PR adds metrics but not tracing yet; is "tracing" the right word here?

We first need to install opentelemetry:

.. code-block:: sh

    pip install opentelemetry-api
    pip install opentelemetry-sdk
    pip install opentelemetry-exporter-google-cloud

(Optionally) You can install this package to augment your metrics with resource info.

.. code-block:: sh

    pip install opentelemetry-tools-google-cloud

Then, at the beginning of your application create a MeterProvider. Example:

.. code:: python

    # "opt-in" for opentelemtry metrics capturing.

>     # Requires you to have installed opentelemetry packages and called set_meter_provider

Reviewer 2: ```suggestion
    # Requires you to have installed opentelemetry packages and called set_meter_provider.
```

    from opentelemetry import metrics
    from opentelemetry.sdk.metrics import MeterProvider

    #==============================================================================

>     # OPTIONAL: These lines of code will detect + scrape resource info from the env/metadata.

Reviewer 2: We can probably shorten this a bit
```suggestion
    # OPTIONAL: These lines of code will scrape resource info from the env/metadata.
```

>     # This info will then be passed down the line and added to metric info

Reviewer 2: ```suggestion
    # This info will then be passed down the line and added to metric info.
```

    from opentelemetry.tools.resource_detector import GoogleCloudResourceDetector
    resources = GoogleCloudResourceDetector().detect()

    # Note that this will NOT work if you are on a cloudtop, as you are on a
    # VM instance that belongs to the project "cloudtop:prod" which you likely

>     # don't have Cloud Monitoring write access to

Reviewer 2: ```suggestion
    # don't have Cloud Monitoring write access to.
```

    #==============================================================================

>     # If you don't want to capture resources, remove resource=resources

Reviewer 2: ```suggestion
    # If you don't want to capture resources, remove 'resource=resources'.
```

    metrics.set_meter_provider(MeterProvider(stateful=False, resource=resources))

>     # Normal code below. This is your application / script

Reviewer 2: ```suggestion
    # Normal code below. This is your application / script.
```
